### ISSUE TITLE
Support for obspy Catalog and Inventory objects

### ISSUE THREAD

**Issue by d-chambers** [jkmacc-LANL/pisces#18]
Hello,

Thank you for sharing this package, it looks awesome. 

Does pisces support obspy Catalog objects (for event data) and obspy Inventory objects (for station data)? I am imagining an API similar to the obspy fdsn client class. For example, the event functionality might look like the following:

```python
import pisces
catalog = pisces.get_events(minmagnitude=4)  # pull events out of the db
for event in catalog:
    make_some_changes(event)  # process the events in some way
pisces.put_events(catalog)  # update all of the modified events in the database
````

Thanks again.

**Comment by jkmacc-LANL** [comment]
Thanks, @d-chambers !  The current answer is "no", but I definitely want to implement an Obspy-like `Client` , where `c.get_events` returns a `Catalog`,  `c.get_stations` returns an `Inventory`, and `c.get_waveforms` returns a `Stream`.  I'm mostly sure that ObsPy Clients have an API that looks like the FDSN-WS spec, but I don't currently have the know-how to implement a `Client` that *fully* follows that spec.  Assembling `Catalog` and `Inventory` objects from database queries is something that needs to happen, though.  PRs are encouraged! :smile:

**Comment by jkmacc-LANL** [comment]
@klaaur Do you have any progress on this?😋

**Comment by klaaur** [comment]
Yes, we ended up not getting an intern so it’s still in the same state. Will check with some people at SNL about sharing and keep you posted. I should push another thing to your git page to inform you that you can no longer import obspy.taup in your util.py file in obspy 1.1.0- haven’t looked up the correct call yet but should be easy to track down.

**Comment by jkmacc-LANL** [comment]
Thanks for looking into it.  Re: obspy.taup, I just pushed a Python 3 version that includes [the taup fix](https://github.com/jkmacc-LANL/pisces/commit/fb3e40321dabdbea3dd85ecb498423cbb357a3b5#diff-6d287414c6d291a8a703269f68904716).  Downside (upside): the new version is Python 3 only.

**Comment by zvikfir** [comment]
This feature will be helpful for me as well. Is there any chance that there's been a progress on this one? I couldn't find it in the docs.

I might be able to help if needed.

**Comment by jkmacc-LANL** [comment]
@rigotibi I know you were interested in this topic. I just want to tag you on it here for visibility.

**Comment by jkmacc-LANL** [comment]
@zvikfir I'm sorry for the delay in responding.  There isn't anything new on this topic, except that there are a number of interested parties.  Thank you for your offer to contribute.  Are you interested in the `Catalog` or `Inventory` objects, or both?

**Comment by jkmacc-LANL** [comment]
Work continues on this in #65
